### PR TITLE
[stat var hierarchy] search updates

### PR DESCRIPTION
**Pull request by chejennifer**
- add exact dcid to search index
- use stat var definition to calculate number of PV for a stat var for ranking

### CODE REVIEW

File: internal/server/resource/resource.go [hunk 1]
```diff
@@ -25,6 +25,18 @@ import (
 // PVs to a number greater than max number of PVs for a human curated stat var.
 const nonHumanCuratedNumPv = 30
 
+// prefixes for pv definitions for required properties
+var requiredPVDefPrefixes = map[string]struct{}{
```
Comment (pradh) [inline internal/server/resource/resource.go:29]: nit: These are all technically not "required" (eg mq, md).  Might rename as "structured" or "known", or something

File: internal/server/resource/resource.go [hunk 2]
```diff
@@ -103,13 +121,28 @@ func (index *SearchIndex) Update(
...
+	if len(svDefinition) > 0 {
+		svDefParts := strings.Split(svDefinition, ",")
+		for _, defPart := range svDefParts {
+			// don't include required properties when counting PVs
+			if len(defPart) >= requiredPVDefPrefixLength {
+				prefix := defPart[0:requiredPVDefPrefixLength]
+				if _, ok := requiredPVDefPrefixes[prefix]; ok {
+					continue
+				}
+			}
```
Comment (pradh) [inline internal/server/resource/resource.go:134]: Among the cases where num-pvs match, we should pick the one with fewer known props.

For example, pt=Person,mp=count,st=measuredValue should come before pt=Person,mp=count,st=measuredValue,md=Area

**Review by pradh — APPROVED** [review]
Nice

File: internal/server/resource/resource.go [hunk 3]
```diff
@@ -103,13 +121,28 @@ func (index *SearchIndex) Update(
 			}
 		}
 	}
-	approxNumPv := len(strings.Split(nodeID, "_"))
-	if approxNumPv == 1 {
-		// when approxNumPv is 1, most likely a non human curated PV
-		approxNumPv = nonHumanCuratedNumPv
+	numPV := 0
+	if len(svDefinition) > 0 {
```
Comment (shifucun) [inline internal/server/resource/resource.go:125]: this is nice

File: internal/server/resource/resource.go [hunk 4]
```diff
@@ -76,12 +88,18 @@ type RankingInfo struct {
...
 func (index *SearchIndex) Update(
-	nodeID string, nodeString string, displayName string, isSvg bool, synonymMap map[string][]string) {
+	nodeID string, nodeString string, displayName string, isSvg bool, synonymMap map[string][]string, svDefinition string) {
 	processedNodeString := strings.ToLower(nodeString)
 	processedNodeString = strings.ReplaceAll(processedNodeString, ",", " ")
 	tokenList := strings.Fields(processedNodeString)
 	// Create a map of tokens/synonyms to the matching string from nodeString
 	tokens := map[string]string{}
+	// add nodeID as a token, but only set the matching string if nodeID is in
+	// the nodeString
```
Comment (shifucun) [inline internal/server/resource/resource.go:98]: Can you add an example here?

**Review by shifucun — APPROVED** [review]

File: internal/server/resource/resource.go (hunk 1 above)
Comment (chejennifer) [reply internal/server/resource/resource.go]: done

File: internal/server/resource/resource.go (hunk 2 above)
Comment (chejennifer) [reply internal/server/resource/resource.go:141]: done

File: internal/server/resource/resource.go (hunk 3 above)
Comment (chejennifer) [reply internal/server/resource/resource.go:132]: thanks!

File: internal/server/resource/resource.go (hunk 4 above)
Comment (chejennifer) [reply internal/server/resource/resource.go:100]: done

**Review by pradh — APPROVED** [review]
Thanks for the changes!

**Pull request merged**